### DISCUSSION TITLE
shouldn't non-basic key pallets be disabled from TD configuration GUI if only basic keycodes are supposed to work with TD?

### DISCUSSION THREAD

**Discussion by TadayukiOkada** [vial-kb/vial-gui#233] [Q&A] [answered]
According to https://docs.qmk.fm/#/feature_tap_dance, you can only use basic keycodes with TD. If so, the GUI shouldn't display non-basic key pallets for TD configuration, should it?
I used grave escape from quantum pallet in a TD and got continuously repeating Esc input when I hit the key. I thought it was a bug in the FW, but it seems that configuration shouldn't be allowed, no?

**Comment by xyzz** [comment, marked as answer]
Vial is intended to support all keycodes with its tap-dance implementation. If you run into an issue with it, you should provide more details like your full keymap. But it certainly works with stuff like layer keys, etc.

**Reply by TadayukiOkada** [reply]
I see. So Vial has diverged from QMK in the TD implementation, I suppose? I have a Sofle variant from IFKB and had the issue (repeating Esc input). I contacted IFKB about the issue as they don't provide the source.

**Reply by xyzz** [reply]
Not necessarily diverged, rather the restriction you saw about basic keycodes applies specifically to ACTION_TAP_DANCE_DOUBLE/ACTION_TAP_DANCE_LAYER_MOVE/etc macros. However these are just helpers in qmk, whereas the underlying tap-dance functionality can do a lot more with custom code, which is what Vial utilizes to provide access to every keycode.
